index.js

```diff
@@ -8,92 +8,71 @@ const RedisStore = require('./store/RedisStore')
 
 const defaultHook = 'onRequest'
 
-async function fastifyRateLimit (fastify, settings) {
-  let labels = {
-    rateLimit: 'x-ratelimit-limit',
-    rateRemaining: 'x-ratelimit-remaining',
-    rateReset: 'x-ratelimit-reset',
-    retryAfter: 'retry-after'
-  }
+const defaultHeaders = {
+  rateLimit: 'x-ratelimit-limit',
+  rateRemaining: 'x-ratelimit-remaining',
+  rateReset: 'x-ratelimit-reset',
+  retryAfter: 'retry-after'
+}
 
-  const draftSpecHeaders = {
-    rateLimit: 'ratelimit-limit',
-    rateRemaining: 'ratelimit-remaining',
-    rateReset: 'ratelimit-reset',
-    retryAfter: 'retry-after'
-  }
+const draftSpecHeaders = {
+  rateLimit: 'ratelimit-limit',
+  rateRemaining: 'ratelimit-remaining',
+  rateReset: 'ratelimit-reset',
+  retryAfter: 'retry-after'
+}
 
-  // create the object that will hold the "main" settings that can be shared during the build
-  // 'global' will define, if the rate limit should be apply by default on all route. default : true
+async function fastifyRateLimit (fastify, settings) {
   const globalParams = {
     global: (typeof settings.global === 'boolean') ? settings.global : true
   }
 
   if (typeof settings.enableDraftSpec === 'boolean' && settings.enableDraftSpec) {
     globalParams.enableDraftSpec = true
-    labels = draftSpecHeaders
+    globalParams.labels = draftSpecHeaders
   } else {
     globalParams.enableDraftSpec = false
+    globalParams.labels = defaultHeaders
   }
 
   globalParams.addHeaders = Object.assign({
-    [labels.rateLimit]: true,
-    [labels.rateRemaining]: true,
-    [labels.rateReset]: true,
-    [labels.retryAfter]: true
+    [globalParams.labels.rateLimit]: true,
+    [globalParams.labels.rateRemaining]: true,
+    [globalParams.labels.rateReset]: true,
+    [globalParams.labels.retryAfter]: true
   }, settings.addHeaders)
 
   globalParams.addHeadersOnExceeding = Object.assign({
-    [labels.rateLimit]: true,
-    [labels.rateRemaining]: true,
-    [labels.rateReset]: true
+    [globalParams.labels.rateLimit]: true,
+    [globalParams.labels.rateRemaining]: true,
+    [globalParams.labels.rateReset]: true
   }, settings.addHeadersOnExceeding)
 
-  globalParams.labels = labels
-
-  // define the global maximum of request allowed
+  // Global maximum allowed requests
   globalParams.max = ((typeof settings.max === 'number' && !isNaN(settings.max)) || typeof settings.max === 'function')
     ? settings.max
     : 1000
 
-  // define the global Time Window
+  // Global time window
   globalParams.timeWindow = typeof settings.timeWindow === 'string'
     ? ms.parse(settings.timeWindow)
     : typeof settings.timeWindow === 'number' && !isNaN(settings.timeWindow)
       ? settings.timeWindow
       : 1000 * 60
-
-  globalParams.timeWindowInSeconds = (globalParams.timeWindow / 1000) | 0
+  globalParams.timeWindowInSeconds = Math.floor(globalParams.timeWindow / 1000)
```

Comment:
Is there a chance to get a number bigger than 2^32?

Reply:
Good question, since it is a user parameter and Number is up to 2^64... it could be IMO